src/parser/jobs/nin/modules/Ninki.tsx

```diff
@@ -1,178 +1,108 @@
 import {ActionLink} from 'components/ui/DbLink'
-import TimeLineChart from 'components/ui/TimeLineChart'
-import {ActionKey} from 'data/ACTIONS'
 import JOBS from 'data/JOBS'
-import {CastEvent} from 'fflogs'
-import Module, {dependency, DISPLAY_MODE} from 'parser/core/Module'
-import {LegacyComboEvent} from 'parser/core/modules/Combos'
+import {Cause, Event, Events} from 'event'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
 import {Data} from 'parser/core/modules/Data'
-import {NormalisedDamageEvent} from 'parser/core/modules/NormalisedEvents'
+import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
 import Suggestions, {Suggestion, TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
-import React, {Fragment} from 'react'
+import React from 'react'
 
-// Constants
-const MAX_NINKI = 100
-
-const GCD_NINKI_GAIN = 5
-const FINISHER_NINKI_GAIN = 10
-const MUG_NINKI_GAIN = 40
-const MEISUI_NINKI_GAIN = 50
-const BUNSHIN_NINKI_GAIN = 5
-const SPENDER_COST = 50
-
-const ACTION_NINKI_GAIN: Array<[ActionKey, number]> = [
-	['SPINNING_EDGE', GCD_NINKI_GAIN],
-	['GUST_SLASH', GCD_NINKI_GAIN],
-	['DEATH_BLOSSOM', GCD_NINKI_GAIN],
-	['HAKKE_MUJINSATSU', GCD_NINKI_GAIN],
-	['THROWING_DAGGER', GCD_NINKI_GAIN],
-	['AEOLIAN_EDGE', FINISHER_NINKI_GAIN],
-	['ARMOR_CRUSH', FINISHER_NINKI_GAIN],
-	['SHADOW_FANG', FINISHER_NINKI_GAIN],
-	['MUG', MUG_NINKI_GAIN],
-	['MEISUI', MEISUI_NINKI_GAIN],
-]
-
-const NINKI_GCDS: ActionKey[] = [
-	'SPINNING_EDGE',
-	'DEATH_BLOSSOM',
-	'THROWING_DAGGER',
-	'SHADOW_FANG',
-]
-
-const NINKI_COMBOS: ActionKey[] = [
-	'GUST_SLASH',
-	'AEOLIAN_EDGE',
-	'ARMOR_CRUSH',
-	'HAKKE_MUJINSATSU',
-]
-
-const NINKI_OGCDS: ActionKey[] = [
-	'MUG',
-	'MEISUI',
-]
-
-const NINKI_SPENDERS: ActionKey[] = [
-	'HELLFROG_MEDIUM',
-	'BHAVACAKRA',
-	'BUNSHIN',
-]
+type GaugeModifier = Partial<Record<Event['type'], number>>
 
-interface DataPoint {
-	t: number,
-	y: number,
+// Constants
+const GCD_GAIN = 5
+const FINISHER_GAIN = 10
+const MUG_GAIN = 40
+const MEISUI_GAIN = 50
+const BUNSHIN_GAIN = 5
+const SPENDER_COST = -50
+
+const OVERCAP_SEVERITY = {
+	20: SEVERITY.MINOR,
+	50: SEVERITY.MAJOR,
 }
 
-export class Ninki extends Module {
-	static override handle = 'ninki'
+export class Ninki extends CoreGauge {
 	static override title = t('nin.ninki.title')`Ninki Timeline`
-	static override displayMode = DISPLAY_MODE.FULL
 
 	@dependency private data!: Data
 	@dependency private suggestions!: Suggestions
 
-	private actionNinkiGain = new Map(ACTION_NINKI_GAIN.map(
-		pair => [this.data.actions[pair[0]].id, pair[1]]
-	))
-
-	private ninkiGcds: number[] = NINKI_GCDS.map(key => this.data.actions[key].id)
+	private ninkiGauge = this.add(new CounterGauge({
+		chart: {label: 'Ninki', color: JOBS.NINJA.colour},
+	}))
+
+	private ninkiModifiers = new Map<number, GaugeModifier>([
+		// Builders
+		[this.data.actions.SPINNING_EDGE.id, {action: GCD_GAIN}],
```

Comment:
Builders on basic weaponskills that don't require a combo effect do still require a successful hit, and should be checking `damage` (and verifying that the damage event represents a successful hit via the isSuccessfulHit utility).  See also Shadow Fang, Death Blossom, Throwing Dagger, and Mug below.  (Though I'm not 100% sure on Mug since it's an Ability... the easiest way to test it ingame currently is to attack the wrong target in a firewall situation like Hades fire/ice adds or O12S M/F).

Reply:
I'm going to operate on the assumption that Mug requires a successful hit for now; it's very likely the correct assumption, and I'm not in a position to test it at the moment anyway. We can revisit for EW (or if someone is able to do that test for me real quick).